Gems/Atom/RPI/Code/Source/RPI.Public/MeshDrawPacket.cpp

```diff
@@ -391,8 +397,8 @@ namespace AZ
                 {
                     AZ_Error(
                         "MeshDrawPacket",
-                        (!m_rootConstantsLayout && !pipelineStateDescriptor.m_pipelineLayoutDescriptor->GetRootConstantsLayout()) ||
-                        (m_rootConstantsLayout && m_rootConstantsLayout->GetHash() == pipelineStateDescriptor.m_pipelineLayoutDescriptor->GetRootConstantsLayout()->GetHash()),
+                        (!m_rootConstantsLayout && !HasRootConstants(rootConstantsLayout)) ||
```

Comment:
I think this condition should be an assert since it may only fail if there is an implementation issue

Reply:
I think it's kind of a grey area depending on if you consider a shader more 'code' or 'data/asset'.

Reply:
I mean this part is implementation issue `(!m_rootConstantsLayout && !HasRootConstants(rootConstantsLayout)) `. I dont think it depends on shader. 
The second part is an error '(m_rootConstantsLayout && m_rootConstantsLayout->GetHash() == pipelineStateDescriptor.m_pipelineLayoutDescriptor->GetRootConstantsLayout()->GetHash()'

Reply:
That case will be hit if the first DrawItem has no root constant in the shader (m_rootConstantsLayout will be null), but a subsequent DrawItem does, so that can still come from the shader.